pkg/ebpf/c/bpf_helpers.h

```diff
@@ -84,6 +84,17 @@ static int (*bpf_probe_read_user)(void* dst, int size, void* unsafe_ptr) = (void
 static int (*bpf_probe_read_kernel)(void* dst, int size, void* unsafe_ptr) = (void*)BPF_FUNC_probe_read_kernel;
 #endif
 
+/* bpf_probe_read_from_user is an helper to use bpf_probe_read_user() on kernel >= 5.5.0
+ * but will use bpf_probe_read() on older and prebuild build.
+ */
+#ifdef DD_PREBUILD
```

Comment:
I think you could do something simpler, and more transparent like:
```
#if defined(DD_PREBUILD) || LINUX_VERSION_CODE < KERNEL_VERSION(5, 5, 0))
static int (*bpf_probe_read_user)(void* dst, int size, void* unsafe_ptr) = (void*)BPF_FUNC_probe_read;
static int (*bpf_probe_read_kernel)(void* dst, int size, void* unsafe_ptr) = (void*)BPF_FUNC_probe_read;
#else
static int (*bpf_probe_read_user)(void* dst, int size, void* unsafe_ptr) = (void*)BPF_FUNC_probe_read_user;
static int (*bpf_probe_read_kernel)(void* dst, int size, void* unsafe_ptr) = (void*)BPF_FUNC_probe_read_kernel;
#endif
```
and then use `bpf_probe_read_user` everywhere rather than a new function.

Reply:
Alternatively, we can try upstreaming a transparent replacement at the loader level in `cilium/ebpf` like what `libbpf` does.

Reply:
cilium/ebpf load will fixup bpf_probe_read_{user,kernel} to bpf_probe_read if kernel doesn't support _{user,kernel}
so the fix will be to rename all our bpf_probe_read(), the loader will do the fixup https://github.com/cilium/ebpf/blob/master/linker.go#L196

Reply:
Oh nice! I didn't realize that was already there.

Reply:
Nevermind, they removed the check for `kern_version` in 5.0

Reply:
I seen that libbpf do the same kind of fixup